lib/cluster.rb

```diff
@@ -169,6 +177,19 @@ def add_members_from(cluster)
     end
   end
 
+  def commitments?(include_deprecated: false)
+    if include_deprecated
+      commitments.any?
+    else
+      commitments.reject(&:deprecated?).any?
+    end
+  end
+
+  def eligible_for_commitments?
+    # be an spm with items and 1+ CH holdings
+    format == "spm" && ht_items.any? && holdings.any? && ch_counts.values.sum > 0
```

Comment:
Is `ch_counts.values.sum > 0` redundant with `holdings.any?` -- that is, could `ch_counts.values.sum > 0` be true if `!holdings.any?`?

In order to make this easier to read, I'd consider extracting `ch_counts.values.sum > 0` as:

```ruby
def any_current_holdings?
  ch_counts.values.sum > 0
end
```

Reply:
I figured if `holdings.any? == false` we short circuit out and don't have to go into `ch_counts`. Haven't done the benchmark to back this up, tho.

Reply:
Hmm.. If `holdings.any? == false` then it seems like `holdings_by_org` will also be empty (and likewise fast since `holdings.group_by(:&organization)` is just running on an empty array), so it shouldn't really be computing anything extra. Not really a big deal either way, but I'd still lean towards removing the check for `holdings.any?`.

Reply:
modified as suggested